### PR TITLE
Allow simultaneous named and default imports

### PR DESCRIPTION
**Pull request by jeffmo**
Previously Flow considered `import Foo from ...` and `import Foo, {Bar, Baz} from ...` to be mutually exclusive -- though they should not be (per spec).

This fixes Flow to properly handle `import Foo, {Bar, Baz} from ...`

### CODE REVIEW

File: src/typing/type_inference_js.ml [hunk 1]
```diff
@@ -1220,46 +1220,53 @@ and statement_decl cx = Ast.Statement.(
...
-            ) in
-            List.iter init_specifier named_specifiers
-          | Some(ImportDeclaration.NameSpace(_, (loc, local_ident))) ->
-            let local_name = local_ident.Ast.Identifier.name in
-            let reason =
-              mk_reason (spf "%s * as %s" import_str local_name) loc
+          | None ->
+            failwith (
+              "Parser error: Non-default imports must have a set of " ^
+              "named import specifiers!"
```
Comment (samwgoldman) [inline src/typing/type_inference_js.ml]: Isn't it possible to import for side effects only, without specifying any bindings, default or non? See #574 for an issue, and [the spec at 15.2.2; `import ModuleSpecifier`](http://www.ecma-international.org/ecma-262/6.0/#sec-imports).

**Comment by samwgoldman** [comment]
Cool! This is a simple enough diff, and clearly does what it says on the tin. For tracking purposes, this fixes #525.

File: src/typing/type_inference_js.ml (hunk 1 above)
Comment (jeffmo) [inline src/typing/type_inference_js.ml]: Oh yea, forgot about that issue. Will fix now

**Comment by nmn** [comment]
Just want to report in saying that I merged this temporarily on my local repo and it works as intended for me. Will run the full test suite and report back.

**Pull request closed without merging**